src/BackgroundTask/CronBasedBackgroundTask.php

```diff
@@ -72,4 +74,23 @@ abstract protected function get_event_name();
 	 * @param mixed[] ...$args Args to pass to the process callback.
 	 */
 	abstract public function process( ...$args );
+
+	/**
+	 * Add new interval for cron schedules.
+	 *
+	 * @param array $cron_schedules List of cron schedules.
+	 *
+	 * @return array List of cron schedules.
+	 */
+	public function add_cron_schedules( $cron_schedules ) {
+
+		$cron_schedules = ( ! empty( $cron_schedules ) && is_array( $cron_schedules ) ) ? $cron_schedules : [];
```

Comment:
Is there a known case where `$cron_schedules` is not an array?

```suggestion
		$cron_schedules = is_array( $cron_schedules ) ? $cron_schedules : [];
```